[194] Logo / Favicon

orenyk opened the pull request:
Resolves #194, pending resolution of hover effects and CSS positioning janky-ness.

mnquintana commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -9,7 +9,7 @@
       </button>
       <% unless @app_configs.nil? || @app_configs.site_title.nil? %>
-        <%= link_to sanitize("<i class=\"fa fa-th-list\"></i> #{site_title}"), root_path, class: 'navbar-brand' %>
+        <%= link_to sanitize("#{image_tag('res_logo.png', height: '32', id: 'logo')}<span>#{site_title}</span>"), root_path, class: 'navbar-brand' %>
```
This is kinda jank - it should really be:
```erb
<%= link_to root_path, class: 'navbar-brand' do %>
  <%= image_tag 'res_logo.png', height: '32', id: 'logo' %>
  <span>#{site_title}</span>
<% end %>
```

orenyk replied:
Haha kinda jank is generous :stuck_out_tongue:. I dropped it in the way we had it, but you're totally right; updating now...

mnquintana commented:
Did you colorize the background of the logo? It should really be transparent.

orenyk commented:
I did, partially because the original illustrator file wasn't set up in a real "vector" way. I'm going to just have to recreate it and then have one will full transparency.

orenyk commented:
Blergh, this took a lot longer than I thought it would. Here's the new logo (Illustrator is a pain in the butt):

![favicon-194x194](https://cloud.githubusercontent.com/assets/901808/6500362/c934153e-c2d8-11e4-9ad6-1109bd01e6c3.png)

![favicon-16x16](https://cloud.githubusercontent.com/assets/901808/6500363/d8c8c526-c2d8-11e4-9ce1-bc844a334022.png)

Now everything's that's supposed to be transparent is transparent, so it should be pretty straightforward to regenerate the navbar logo.

orenyk commented:
Here's the updated logo:

![res_logo](https://cloud.githubusercontent.com/assets/901808/6500424/7e49e7fa-c2d9-11e4-8649-0fd9d2b90ce2.png)

orenyk commented:
I'll get these into my branch tomorrow and make sure everything plays nicely.

mnquintana commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -9,7 +9,10 @@
-        <%= link_to sanitize("<i class=\"fa fa-th-list\"></i> #{site_title}"), root_path, class: 'navbar-brand' %>
+        <%= link_to root_path, class: 'navbar-brand' do %>
+          <%= image_tag 'res_logo.png', height: '32' %>
+          <span><%= "#{site_title}" %></span>
```
Why is this an interpolated string? It could just be `<%= site_title %>`

orenyk commented:
Oh man I haven't done ERB stuff in so long, my bad.

orenyk commented:
Tried dealing with the hover effect, major pain in the butt. See #194 for more details, tried both recoloring SVG's and CSS / div backgrounds but couldn't get it to work properly. Ended up going with a basic jQuery `hover` function, any suggestions for improvement are welcome.

Also downsized the logo images, but then realized we might want higher resolution files for high DPI screens or whatever... might be beyond the capabilities of our app though :stuck_out_tongue:. Let me know how it looks!

mnquintana commented:
So IMO, we shouldn't use an image for this, but rather one of two things:
- an SVG
- an icon font

(For the latter we'd have to make one.) One of the reasons I chose an icon from FontAwesome as the placeholder logo was so it'd be resolution-independent – going back to ol' raster images is a bit of a downgrade.

We would then handle the color and sizing of the logo in our CSS, instead of it being "hard-coded" in an image file. Does that make sense?

I'd also like to add the source files (Illustrator file?) to the repo too (not sure where exactly...)

orenyk commented:
I agree, that was my first try but it didn't work because we have to filled paths (the "R" and the area above it) and the rest are transparent filled. I could try going back into the Illustrator file to split the remaining path; in that case I could go back to my CSS-driven SVG changes vs two separate image files. I guess a second option is just switching to two SVG's instead of two PNGs.

I've got all the source files and will find a place with them in the repo.

mnquintana commented:
What about an icon font?

orenyk commented:
Tried creating an icon font but because we have some unfilled paths the online generator I found didn't work. I switched to using two SVG's instead and some basic JS to switch the `src` files on hover; it can be a little slow at first but seems to be adequate in my mind. I also uploaded the *.ai file to the repository (stuck it in `app/assets/images/`). Let me know what you think!

orenyk commented:
Update: trying to use the inline SVG method I [mentioned]() in the issue still doesn't work because the SVG export from Illustrator ends up creating some closed paths (like the clock outline) that should really be open; this means that changing the fill color in the whole SVG ends up destroying the clock :confused:. I'd really rather use CSS to switch the logo color but I'm not sure how to do that right now :cry:.

squidgetx commented:
I used the `inline svg` gem to, well, inline the svg so that we can use CSS/JS to change the color instead of having to replace the image source, speeding up the animation by quite a bit (and removing the need for having two different svg logo files)

mnquintana commented:
@squidgetx Why'd you take the Illustrator file out?

squidgetx commented:
I figured it wasn't necessary since the current .svg contains enough information. I can put it back if you want to mess with it but the current .svg can be colored entirely through setting fill and stroke anyway

orenyk commented:
Of course there's a gem to do what I was doing manually before :-P. I'm with @mnquintana, let's keep the AI source file in case anyone ever wants to edit it using the native Illustrator format (I'm aware that the SVG should have enough information but the paths are slightly different in the Illustrator file). Reading through the code now...

orenyk commented on app/assets/javascripts/application.js:
```diff
@@ -71,6 +71,15 @@ $(document).on('railsAutocomplete.select', '#fake_searched_id', function(){
 
 $(document).ready(function() {
 
+  // Logo Swap
```
So this is still a JS swap, not a CSS swap? That was the issue that I ran into before; while having an inline SVG allowed us to manipulate its colors using CSS / JS, I couldn't figure out how to do it with just CSS because of the way that Illustrator exported the SVG file. In retrospect, this is still cleaner than swapping files, so it's probably the best solution.

orenyk commented:
Ok, I made some whitespace fixes, re-added the AI file, and slightly increased the right padding on the logo. This should be good to go, @mnquintana do you want to give it one more look over and then I'll squash / merge? Thanks @squidgetx for the cleaner inline SVG implementation!

squidgetx commented:
I figured out how to do the change using CSS `inherit` so we don't even need any JS anymore :D

orenyk commented:
Awesome, great job! I renamed the logo file to `res_logo.svg` and I think this is good to go. @squidgetx, are you good for me to squash / merge?

squidgetx commented:
:shipit:

mnquintana commented:
@orenyk Let me take a look at this tonight before you do

mnquintana commented on app/assets/stylesheets/layouts/_navbar.css.scss:
```diff
@@ -43,18 +43,33 @@
+  stroke: #FAFDFE;
+}
+
+#logo {
```
Why is there both a class `logo` and an id `logo`? That is super confusing. This should be a more descriptive, namespaced name and a class instead of an id (same of the other `.logo`)

orenyk replied:
Would something like `.logo-container` for the `div` and `.logo` for the `img` make more sense?

mnquintana commented on app/assets/stylesheets/layouts/_navbar.css.scss:
```diff
@@ -43,18 +43,33 @@
-                 0 0 150px lighten($right-1-color, 20%);
-    // color: lighten($right-1-color, 60%);
-  }
+.navbar-brand:hover{
```
Spacing

mnquintana commented on app/assets/stylesheets/layouts/_navbar.css.scss:
```diff
@@ -112,3 +127,4 @@
 .navbar .btn-sign-in {
 	margin-right: 0px;
```
Would you mind fixing the indentation here? (I know it wasn't changed in the PR, but if we don't know it'll probably never happen)

mnquintana commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -9,7 +9,14 @@
       <% unless @app_configs.nil? || @app_configs.site_title.nil? %>
-        <%= link_to sanitize("<i class=\"fa fa-th-list\"></i> #{site_title}"), root_path, class: 'navbar-brand' %>
+        <%= link_to root_path, class: 'navbar-brand' do %>
+          <div class='logo'>
```
Double quote the attribute value plz

orenyk replied:
Lol, we're getting Ruby in our HTML :stuck_out_tongue:

mnquintana commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -9,7 +9,14 @@
+          <div class='logo'>
+            <%= inline_svg 'res_logo.svg', id: 'logo' %>
+          </div>
+          <div class='title'>
```
Double quote the attribute value plz

orenyk commented:
I dealt with @mnquintana's comments, added a little SASS magic to our stylesheet for the hover styling, and squashed. @mnquintana, want to double check it or am I good to merge?
